pkg/storage/pvc.go

```diff
@@ -150,6 +150,35 @@ func (builder *PVCBuilder) WithStorageClass(storageClass string) (*PVCBuilder, e
 	return builder, nil
 }
 
+// WithVolumeMode configures what type of volume is required by the claim.
+func (builder *PVCBuilder) WithVolumeMode(volumeMode string) (*PVCBuilder, error) {
+	glog.V(100).Infof("Set VolumeMode %s for the PersistentVolumeClaim", volumeMode)
+
+	if volumeMode == "" {
+		glog.V(100).Infof("Empty volumeMode requested for the PersistentVolumeClaim")
+
+		builder.errorMsg = fmt.Sprintf("Empty volumeMode requested for the PersistentVolumeClaim %s",
+			builder.Definition.Name)
+
+		return builder, fmt.Errorf(builder.errorMsg)
+	}
+
+	if !validateVolumeMode(volumeMode) {
+		glog.V(100).Infof(fmt.Sprintf("Unsupported VolumeMode: %s", volumeMode))
+
+		builder.errorMsg = fmt.Sprintf("Unsupoorted VolumeMode %q requested for %s PersistentVolumeClaim",
```

Comment:
misspell: Unsupoorted
Add the namespace

Reply:
thanks, updated